containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/commands/PullImageCommandHandler.java

```diff
@@ -89,19 +89,19 @@ protected IStatus run(final IProgressMonitor monitor) {
 					if (registry == null || registry.isDockerHubRegistry()) {
 						((DockerConnection) connection).pullImage(imageName,
 								new DefaultImagePullProgressHandler(connection,
-										imageName));
+										imageName, monitor));
 					} else {
 						String fullImageName = registry.getServerHost() + '/'
 								+ imageName;
 						if (registry instanceof IRegistryAccount) {
 							IRegistryAccount account = (IRegistryAccount) registry;
 							((DockerConnection) connection).pullImage(fullImageName,
 									account, new DefaultImagePullProgressHandler(
-											connection, fullImageName));
+											connection, fullImageName, monitor));
 						} else {
 							((DockerConnection) connection).pullImage(fullImageName,
 									new DefaultImagePullProgressHandler(connection,
-											fullImageName));
+											fullImageName, monitor));
 						}
 					}
 				} catch (final DockerException e) {
```

Comment:
There needs to be special logic to catch a cancelled exception.  It is not an error and should not be treated the same way.  I don't want to have a blocking dialog come up and say an error occurred because I cancelled a job.